### PR TITLE
feat: POAP Contracts type definitions and interaction scripts

### PR DESCRIPTION
**Pull request by manu0466**
This PR adds the type definionts for the POAP contract available [here](https://github.com/desmos-labs/desmos-contracts) and provides a set of utility scripts to interact with the contract

### CODE REVIEW

File: types/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Types
+Package that provides the type to interact with the smart contracts available [here](https://github.com/desmos-labs/desmos-contracts).
```
Comment (bragaz) [inline types/README.md:2]:
```suggestion
Package that provides the types to interact with the available smart contracts [here](https://github.com/desmos-labs/desmos-contracts).  
```

File: utils/README.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# Utils
+Set of utility functions to interact with the Desmos smart contract available [here](https://github.com/desmos-labs/desmos-contracts).
```
Comment (bragaz) [inline utils/README.md:2]:
```suggestion
Set of utility functions to interact with the Desmos available smart contracts [here](https://github.com/desmos-labs/desmos-contracts).
```

**Review by bragaz — COMMENTED** [review]
I've lost this review days ago but I've found it while looking at the Desmos Jira.
@dadamu please review it as well.

File: types/src/codegen.ts [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+    };
+
+    const compilationSpecs = [];
+    let fileCount = 0;
+    log("Calculating generation specs...");
+    for (const root of config.schemaRoots) {
+        const {name, paths, outputName, outputDir} = root;
+        const contractOutputPath = path.join(outputDir, outputName);
+        // log(`Clearing output path ${contractOutputPath}`)
+        // removeDirectory(contractOutputPath)
```
Comment (dadamu) [inline types/src/codegen.ts:328]: I think it can be removed.

File: utils/upload-contract.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    program
+        .name("Contract uploader")
+        .requiredOption("--contract [contract]", "Path to the contract to upload");
+    program.parse();
+
+    const options = program.opts<ProgramOptions>();
+    // Checks if the smart contract is present.
+    if (!fs.existsSync(options.contract)) {
+        console.error(`Smart contract ${options.contract} don't exist`);
+        return 1;
```
Comment (dadamu) [inline utils/upload-contract.ts:21]:
```suggestion
        return;
```
Function error should be -1 in C. In addition, I think we can just fast return nothing in javascript.

File: utils/upload-contract.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    // Get the account to use to upload the contract code.
+    const account = await signer.getCurrentAccount();
+
+    console.log(`Uploading contract as ${account!.address}`);
+    const uploadResult = await client.upload(account!.address, contractContent, "auto");
+
+    console.log("Contract uploaded");
+    console.log(uploadResult);
+
+    return 0;
```
Comment (dadamu) [inline utils/upload-contract.ts:44]:
```suggestion
```
Return 0 can be removed as the reason above.

File: utils/poap.ts [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+        .description("Initialize a new instance of a poap contract")
+        .requiredOption("--code-id <code-id>", "Id of the contract to initialize", parseInt)
+        .requiredOption("--cw721-code-id <cw721-code-id>", "Id of the cw721 contract that will be initialized from the poap contract", parseInt)
+        .requiredOption("--name <name>", "Poap name")
+        .requiredOption("--symbol <symbol>", "Poap symbol")
+        .requiredOption("--start <start>", "Datetime when the event will start in RFC3339 format (2022-12-31:10:00:00)", parseCWTimestamp)
+        .requiredOption("--end <end>", "Date when the event will end in RFC3339 format (2022-12-31:10:00:00)", parseCWTimestamp)
+        .requiredOption("--address-limit <address_limit>", "Max number of poap that an user can receive", parseInt)
+        .requiredOption("--poap-uri <poap_uri>", "ipfs poap uri that contains the poap metadata", parseIpfsUri)
+        .requiredOption("--event-uri <event_uri>", "ipfs event uri that contains the event metadata", parseIpfsUri)
```
Comment (dadamu) [inline utils/poap.ts:28]: The POAP schema should be updated, newer version has no event-uri field.

File: utils/poap.ts [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+        .action(async (options) => {
+            console.log(`Initializing contract with code ${options.codeId}`);
+            let instantiateMsg: InstantiateMsg = {
+                admin: options.admin,
+                minter: options.minter,
+                cw721_code_id: options.cw721CodeId.toString(),
+                event_info: {
+                    creator: options.creator,
+                    poap_uri: options.poapUri.toString(),
+                    event_uri: options.eventUri.toString(),
```
Comment (dadamu) [inline utils/poap.ts:40]:
```suggestion
```

**Review by dadamu — APPROVED** [review]
LGTM

File: utils/poap.ts [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+
+    const queryCommand = program.command("query")
+        .description("Subcommand to query the contract");
+
+    queryCommand
+        .command("state")
+        .description("Queries the contract state")
+        .requiredOption("--contract <contract>", "bech32 encoded contract address")
+        .action(async (options) => {
+            console.log(`Quering state of ${options.contract}`)
```
Comment (dadamu) [inline utils/poap.ts:63]:
```suggestion
            console.log(`Querying state of ${options.contract}`)
```

**Review by bragaz — APPROVED** [review]
Looks good to me 👍

**Pull request merged**